python/shotgun_globals/date_time.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import datetime
+
+def create_human_readable_timestamp(dt, postfix=""):
+    """
+    Return the time represented by the argument as a string where the date portion is
+    displayed as "Yesterday", "Today", or "Tomorrow" if appropriate.
+
+    By default just the date is displayed, but additional formatting can be appended
+    by using the postfix argument.
+
+    :param dt: The date and time to convert to a string
+    :type dt: :class:`datetime.datetime` or float
+
+    :param postfix: What will be displayed after the date portion of the dt argument
+    :type postfix: A strftime style String
+
+    :returns: A String representing dt appropriate for display
+    """
+    # shotgun_model converts datetimes to floats representing unix time so
+    # handle that as a valid value as well
+    if isinstance(dt, float):
+        dt = datetime.datetime.fromtimestamp(dt)
```

Comment:
A unix timestamp is defined as the number of seconds that have elapsed since 00:00:00 Coordinated Universal Time (UTC), Thursday, 1 January 1970. So I *think* with this code you will be comparing a unix timestamp that is in the UTC timezone with `datetime.datetime.now` in your local timezone. I think `datetime.datetime.fromtimestamp(dt)` creates a a datetime object with "empty" time zone def.

One way to fix this is to use the shotgun UTC timezone object when creating the unix time datetime object. Then I think the code would work. But I might be wrong about all of the above - just wanted to double check - python time zones are so confusing to me. :)